docs/19_feature_design_template.md

```diff
@@ -83,20 +83,89 @@ The above questions are focused on one purpose: the place and role of the new fe
 
 # Design
 
+The design section purposes discovering the implementation details for the feature. What is the structure of the feature? How is the feature to be integrated into the project/component(s)? And many other details are covered by the subsections below. But before we dive into explanations for the design parts, one should consider the following rules that hold for the whole design section:
+
+- The design acts within the requirements examined in the [Requirements](#requirements) section.
+- The design details all the [Analysis]($analysis) section describes/touches.
+- The design's subsections must not leave open questions about the selected implementation approach. Each section ends with clear and definitive details about exactly one implementation approach.
+
 ### Architecture
 > Fundamental structures of the feature and context (diagram).
 
+This section examines the overall, top-level architecture of the feature. This architecture defines the approach to the feature implementation discussed in more detail in the next section. The following points hold for the section:
+
+- The section provides and examines the fundamental architecture of the feature with no details (classes, methods, etc.).
+- The section explains the implementation approach to the feature. If there are several implementation approaches:
+    - the section examines each approach and reviews its pros and cons;
+    - the section selects one approach and explains the choice providing clear reasoning.
+- The section represents the architecture with the components diagram demonstrating responsibilities and the overall algorithm.
```

Comment:
It feels like we need to use something else than the components diagram or let know how it will be different from the component diagram used in the "System modeling" section.

Helpful references could be found here: https://www.alibabacloud.com/blog/how-to-create-an-effective-technical-architectural-diagram_596100 as well as a simple intro to Architecture.